### DISCUSSION TITLE
upload_S3: Unexpected arg kmsEncryptionKey

### DISCUSSION THREAD

**Discussion by superbobdthm** [Velocidex/velociraptor#2694] [Q&A]
I am attempting to update the S3_Upload functionality to allow for specifying the KMS key when `aws:kms` is selected for the server side encryption. In my use case, we have one bucket and instead of specifying a key for the entire bucket, we use different keys for various prefixes of the S3 bucket. 

I found this PR https://github.com/Velocidex/velociraptor/pull/1064/files and thought I could basically duplicate it with the new parameter. 

My initial test case can be found here: https://github.com/superbobdthm/velociraptor/pull/1/files

Everything builds correctly, I can install the server build offline collectors. But when I try to run them I am getting 

```
upload_S3: Unexpected arg kmsEncryptionKey
``` 

Full Log can be found here: [collector_log.txt](https://github.com/Velocidex/velociraptor/files/11533259/collector_log.txt)

If I am reading this error correctly, it seems like the upload_s3 function isn't expecting the kmsEncryptionKey argument, but if you look at the PR. I have included it. 

Is there somewhere else I need to specify the new argument? or additional logging I can get to troubleshoot this error?

**Comment by scudette** [comment]
If you are developing new functionality that should be used by the offline collector you will need to manually upload the offline collector into the tools setup screen - otherwise Velociraptor will just package the release binary and that one does not know about the new parameter.

**Reply by superbobdthm** [reply]
Oh... That makes so much more sense! Thank you. I was loosing my mind trying to understand why this was happening.
